hyperspy/signal.py

```diff
@@ -4039,39 +4057,132 @@ def is_rgb(self):
         >>> import scipy.misc
         >>> im = hs.signals.Signal2D(scipy.misc.ascent())
-        >>> m = hs.plot.markers.rectangle(x1=150, y1=100, x2=400,
+        >>> m = hs.markers.rectangle(x1=150, y1=100, x2=400,
         >>>                                  y2=400, color='red')
         >>> im.add_marker(m)
 
+        Adding to a 1D signal, where the point will change
+        when the navigation index is changed
+        >>> s = hs.signals.Signal1D(np.random.random((3, 100)))
+        >>> marker = hs.markers.point((19, 10, 60), (0.2, 0.5, 0.9))
+        >>> s.add_marker(marker, permanent=True, plot_marker=True)
+        >>> s.plot(plot_markers=True) #doctest: +SKIP
+
+        Add permanent marker
+        >>> s = hs.signals.Signal2D(np.random.random((100, 100)))
+        >>> marker = hs.markers.point(50, 60)
+        >>> s.add_marker(marker, permanent=True, plot_marker=True)
+        >>> s.plot(plot_markers=True) #doctest: +SKIP
+
+        Add permanent marker which changes with navigation position, and
+        do not add it to a current plot
+        >>> s = hs.signals.Signal2D(np.random.randint(10, size=(3, 100, 100)))
+        >>> marker = hs.markers.point((10, 30, 50), (30, 50, 60), color='red')
+        >>> s.add_marker(marker, permanent=True, plot_marker=False)
+        >>> s.plot(plot_markers=True) #doctest: +SKIP
+
+        Add permanent marker without adding it a current plot, and not opening
+        a new plot if none is currently open
+        >>> s = hs.signals.Signal2D(np.random.randint(10, size=(3, 100, 100)))
+        >>> marker = hs.markers.point((10, 30, 50), (30, 50, 60), color='red')
+        >>> s.add_marker(marker, permanent=True,
+        >>>        plot_marker=False, plot_signal=False)
+        >>> s.plot(plot_markers=True) #doctest: +SKIP
+
+        Removing a permanent marker
+        >>> s = hs.signals.Signal2D(np.random.randint(10, size=(100, 100)))
+        >>> marker = hs.markers.point(10, 60, color='red')
+        >>> marker.name = "point_marker"
+        >>> s.add_marker(marker, permanent=True)
+        >>> del s.metadata.Markers.point_marker
         """
-        if self._plot is None:
-            self.plot()
-        if plot_on_signal:
-            self._plot.signal_plot.add_marker(marker)
-        else:
-            self._plot.navigator_plot.add_marker(marker)
-        if plot_marker:
+        marker_data_shape = marker._get_data_shape()
+        if (not (len(marker_data_shape) == 0)) and (
+                marker_data_shape != self.axes_manager.navigation_shape):
+            raise ValueError(
+                    "Navigation shape of the marker must be 0 or the "
+                    "same navigation shape as this signal.")
+        if (marker.signal is not None) and (marker.signal is not self):
+            raise ValueError("Markers can not be added to several signals")
+        marker._plot_on_signal = plot_on_signal
+        marker._plot_marker = plot_marker
+        if plot_signal:
```

Comment:
I didn't want to break the previous default behaviour, which was always to call `s.plot()` if `s._plot()` did not exist. https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/signal.py#L4067

However, I think it is useful to suppress the calling of `s.plot()` for certain cases, for example when running on headless system or returning a signal with permanent marker (like in Atomap).